src/main/java/randoop/generation/Detective.java

```diff
@@ -0,0 +1,332 @@
+                executable instanceof Constructor
+                    ? new ConstructorCall((Constructor<?>) executable)
+                    : new MethodCall((Method) executable);
+
+            NonParameterizedType declaringType = new NonParameterizedType(currentTypeClass);
+            TypedOperation typedClassOperation =
+                new TypedClassOperation(callableOperation, declaringType, inputTypes, outputType);
+
+            // Add the method call to the producerMethodSet.
+            producerMethodSet.add(typedClassOperation);
+            producerTypeSet.addAll(inputTypeList);
+          }
+          processed.add(currentType);
+          // Recursively search for methods that construct objects of the specified type.
+          workList.addAll(producerTypeSet);
+        }
+      }
+    }
+    return producerMethodSet;
+  }
+
+  /**
+   * Given an array of classes, this method converts them into a list of Types.
+   *
+   * @param classes an array of reflection classes
+   * @return a list of Types
+   */
+  private static List<Type> classArrayToTypeList(Class<?>[] classes) {
+    List<Type> inputTypeList = new ArrayList<>();
+    for (Class<?> inputClass : classes) {
+      Type inputType = classToType(inputClass);
+      inputTypeList.add(inputType);
+    }
+    return inputTypeList;
+  }
+
+  /**
+   * Given a class, this method converts it into a Type.
+   *
+   * @param clazz a class
+   * @return a Type
+   */
+  private static Type classToType(Class<?> clazz) {
+    if (clazz.isPrimitive()) {
+      return PrimitiveType.forClass(clazz);
+    } else {
+      return new NonParameterizedType(clazz);
+    }
+  }
+
+  /**
+   * Given a TypedOperation, this method finds a sequence of method calls that can generate an
+   * instance of each input type required by the TypedOperation. It then merges these sequences into
+   * a single sequence.
+   *
+   * @param mainObjPool the main object pool from which to draw input sequences
+   * @param secondaryObjPool the secondary object pool from which to draw input sequences if the
+   *     main object pool does not contain a suitable sequence
+   * @param typedOperation the operation for which input sequences are to be generated
+   * @return a sequence that ends with a call to the provided TypedOperation and contains calls to
+   *     generate each required input, or null if no such sequence can be found
+   */
+  private static @Nullable Sequence getInputAndGenSeq(
+      ObjectPool mainObjPool, ObjectPool secondaryObjPool, TypedOperation typedOperation) {
+    TypeTuple inputTypes = typedOperation.getInputTypes();
+    List<Sequence> inputSequences = new ArrayList<>();
+    List<Integer> inputIndices = new ArrayList<>();
+    Map<Type, List<Integer>> typeToIndex = new HashMap<>();
+
+    // 'index' tracks the global position of each variable across all sequences, used for mapping
+    // variable types to their indices in the final sequence.
```

Comment:
I find this comment confusing.  How does it track "each variable" if it is just one `int`?  The last sentence is also confusing to me.  Could you clarify?

Reply:
Comments modified for getInputAndGenSeq